web3.js/src/connection.ts

```diff
@@ -2054,9 +2112,34 @@ export class Connection {
    */
   async getProgramAccounts(
     programId: PublicKey,
-    commitment?: Commitment,
+    configOrCommitment?: GetProgramAccountsConfig | Commitment,
   ): Promise<Array<{pubkey: PublicKey; account: AccountInfo<Buffer>}>> {
-    const args = this._buildArgs([programId.toBase58()], commitment, 'base64');
+    const extra: Pick<GetProgramAccountsConfig, 'dataSlice' | 'filters'> = {};
+
+    let commitment;
+    let encoding;
+    if (configOrCommitment) {
+      if (typeof configOrCommitment === 'string') {
```

Comment:
This is not super elegant. Maybe there is an alternative.

Reply:
Using typeof is the right approach IMO